docs/src/testing/testing.md

```diff
@@ -81,20 +81,68 @@ Failures:
 Sometimes you want to mark a test as expected to fail. This is useful when you want to verify that an action fails as
 expected.
 
-To mark a test as expected to fail, use the `#[should_panic]` attribute. You can pass the expected failure message as an
-argument to the attribute to verify that the test fails with the expected message
-with `#[should_panic(expected: ('panic message', 'eventual second message',))]`.
+To mark a test as expected to fail, use the `#[should_panic]` attribute.
+
+You can specify the expected failure message in three ways:
+
+1. **With felt**
+   ```rust
+   #[should_panic(expected: 'panic message')]
+   ```
+
+2. **With tuple of felts**:
+   ```rust
+   #[should_panic(expected: ('panic message', 'eventual second message',))]
+   ```
+
+3. **With ByteArray**:
+   ```rust
+   #[should_panic(expected: "panic message")]
+   ```
+   With this format, the expected error message needs to be a substring of the actual error message. This is particularly useful when the error message includes dynamic data such as a hash or address.
```

Comment:
This should be listed as a default option (first in the docs) since it's the most convenient
Also there should be a clear correlation between the type of test that it suits and the example attribute (i would place them side by side for easier information digestion)